lib/gdal/extensions/raster_band_classifier.rb

```diff
@@ -0,0 +1,166 @@
+# frozen_string_literal: true
+
+require 'gdal'
+require 'numo/narray'
+
+module GDAL
+  # Takes a list of Ranges of color values and remaps them.  Note that these
+  # values are directly written to the raster band, overwriting all existing
+  # values.
+  #
+  # @example
+  #   classifier = GDAL::RasterBandClassifier.new(raster_band)
+  #   ranges = [
+  #     { range: 0...20, map_to: 1 },
+  #     { range: 20...50, map_to: 2 },
+  #     { range: 50...250, map_to: 3 }
+  #   ]
+  #   classifier.add_ranges(ranges)
+  #   classifier.classify!(ranges)
+  #
+  # @param ranges [Array<Hash{range => Range, map_to => Number}>]
+  class RasterBandClassifier
```

Comment:
nuke?

Reply:
Only twilight.  Kurt explored using it in FS but instead he made a rust crate.  We could move this into twilight or explore using Kurt's crate.

Reply:
Hm... it seems like a fairly specialized thing, in which case feels like it could exist as a separate gem. ...idk, it does feel like a common operation though. Maybe I'll leave for 1.0, then consider removing for 2.0.